### PR TITLE
disable sending agent startup event when hostname is not set

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

Check if we've managed to retrieve hostname before sending agent startup event.

### Motivation

On ECS Fargate, we cannot get the hostname so we get this text `None's datadog agent started` as an Agent startup event on the Datadog events dashboard. 

### Additional Notes

Should we consider this as a frontend issue and solve it elsewhere not at the Agent level?

### CODE REVIEW

File: pkg/aggregator/aggregator.go [hunk 1]
```diff
@@ -274,7 +274,10 @@ func (agg *BufferedAggregator) AddAgentStartupTelemetry(agentVersion string) {
 		SampleRate: 1,
 		Timestamp:  0,
 	}
-	agg.eventIn <- event
+	if agg.hostname != "" {
```
Comment (clamoriniere) [inline pkg/aggregator/aggregator.go]: you can create the `event` only if the condition is true

Comment (ahmed-mez) [reply pkg/aggregator/aggregator.go]: yes it's the starup event to inform that a Datadog Agent has started on a specific host, its format on the events page is `<hostname>'s Datadog Agent has started`.
For Fargate we set the hostname to `""` which result having `None’s datadog agent started` as an event

Comment (ahmed-mez) [reply pkg/aggregator/aggregator.go]: do you suggest doing that way ?

```golang
if agg.hostname != "" {
    event := metrics.Event{
		Text:           fmt.Sprintf("Version %s", agentVersion),
		SourceTypeName: "System",
		Host:           agg.hostname,
		EventType:      "Agent Startup",
	}
    agg.eventIn <- event
}
```

Comment (clamoriniere) [reply pkg/aggregator/aggregator.go]: yes or can be also

```golang
if agg.hostname != "" {
    agg.eventIn <- metrics.Event{
		Text:           fmt.Sprintf("Version %s", agentVersion),
		SourceTypeName: "System",
		Host:           agg.hostname,
		EventType:      "Agent Startup",
	}
}

```

Comment (ahmed-mez) [reply pkg/aggregator/aggregator.go]: thank you! I've made the change

**Review by clamoriniere — DISMISSED** [review]
/lgtm

File: pkg/aggregator/aggregator.go [hunk 2]
```diff
@@ -272,7 +266,15 @@ func (agg *BufferedAggregator) AddAgentStartupTelemetry(agentVersion string) {
...
+		// Send startup event only when we have a valid hostname
+		agg.eventIn <- metrics.Event{
+			Text:           fmt.Sprintf("Version %s", agentVersion),
+			SourceTypeName: "System",
+			Host:           agg.hostname,
+			EventType:      "Agent Startup",
+		}
+	}
 	agg.metricIn <- metric
```
Comment (arbll) [inline pkg/aggregator/aggregator.go]: Super nit: let's put this under the creation of `metric`

**Review by arbll — DISMISSED** [review]
One nit, lgtm otherwise. Feel free to merge once addressed.

File: pkg/aggregator/aggregator.go (hunk 2 above)
Comment (ahmed-mez) [reply pkg/aggregator/aggregator.go]: done 👍

**Review by clamoriniere — APPROVED** [review]

**Comment by ahmed-mez** [comment]
@arbll can you re approve the PR please?

**Pull request merged**